### ISSUE TITLE
[P1][EPIC] Human Attention Architecture — AI駆動開発全体をHuman Decision Surfaceへ最適化する

### ISSUE THREAD

**Issue by s977043** [s977043/PlanGate#1343]
## Why

AI駆動開発では、AI側の探索・実装・レビュー・検証能力を増やすほど、raw output をそのまま Human へ渡した場合の読解・確認・判断コストが増える。

この問題を River Review のUI改善だけに閉じず、PlanGate / ai-dev / ai-loop V2 / Evolution をまたぐ **Human Attention Architecture** として扱う。

North Star:

> **Machine complexity should not become Human complexity.**

共通原則:

1. **Think deeply, materialize selectively**
2. **Visibility is complete, attention is selective**
3. **Compress before escalating**
4. **Escalate decisions, not process logs**

Human Attention削減は Human visibility / safety / correctness の削減を意味しない。

---

# Scope

本Epicは同じ機能を各レイヤへコピーしない。

```text
Shared Principle
      ↓
Layer-specific Decision Surface
```

| Layer | Surface | Responsibility |
| --- | --- | --- |
| River Review | Review Decision Surface | findings / coverage / resolution を人間判断へ投影 |
| ai-loop V2 | Loop Decision Surface | lifecycle / outcome / stop / policy を人間判断へ投影 |
| ai-dev | PR_CREATED Handoff Surface | 実装完了時の handoff cost を削減 |
| PlanGate Plan | Plan Decision Surface | Planのmaterial decisionのみをHumanへ提示 |
| Evolution | Promotion Surface | Harness改善候補をHuman判断へ圧縮 |

---

# Existing boundaries / do not break

## ai-loop V2

既存 North Star / Phase 0 migration を尊重する。

- `ai-dev-workflow` public contract remains Stable
- Legacy `ai-loop` is frozen / PoC
- 新規機能は ai-loop V2 側へ
- Human-owned authority remains human-owned
- Delivery / Evolution は別 state / gate / success condition
- Worker != Verifier != Decision Engine
- hidden CoT / raw session transcript を資産化しない

## PlanGate Plan

s977043/PlanGate#1337 が #1335 Plan Design Principles の paired evaluation 中。

**#1337 の結果固定前に ai-dev-plan Skill / Plan Design Principles candidate を変更しない。**

## River Review

最初の実証環境として以下を利用する。

- s977043/river-review#2368 Human Attention Architecture
- PR #2369 ADR-012
- issue #2370 Decision Surface renderer
- PR #2371 implementation

River Review の結果を横断展開の Evidence として扱う。

---

# Architecture

```text
Machine work
  ├─ Explore
  ├─ Plan
  ├─ Build
  ├─ Review
  ├─ Verify
  ├─ Diagnose
  └─ Evolve
         ↓
Synthesis / Projection
         ↓
Human Decision Surface
         ↓
Human Decision
```

Human Decision Surface は新しい Judge ではない。

```text
Projection != Judgment
Compression != Suppression
Summary != SSoT
```

---

# Workstreams

## WS0 — Canon / Shared Principle

Human Attention Principle を ai-loop V2 canon と整合させる。

- existing North Starとの重複 / 矛盾を gap analysis
- Human Attentionを独立artifact / subsystemとして作らない
- Context compression / handoff / human intervention metricsとの関係を固定
- layer-specific ownershipを固定

### Exit

共通原則と責務境界が文書化され、各WSがそれを参照できる。

---

## WS1 — River Review empirical proof

Owner repo: `s977043/river-review`

- Decision Surface
- all-findings visibility
- incomplete coverage visibility
- Human Attention metrics
- dogfood / baseline comparison

### Exit

最低条件:

```text
Human Attention decreases

AND

Critical/Major visibility >= baseline
False suppression = 0
Provenance loss = 0
Incomplete coverage visibility = 100%
Human reverse rate <= baseline
```

---

## WS2 — ai-loop V2 Loop Decision Surface

Legacy ai-loopへ追加しない。

既存V2 stateを projection する。

Inputs:

- Lifecycle State
- Terminal Outcome
- Stop Reason
- Policy Verdict
- VerificationResult
- FailureRecord
- RunEvidence

Human-facing outputs例:

- MERGE_READY
- HUMAN DECISION REQUIRED
- BLOCKED
- WAITING_EXTERNAL
- verification uncertainty

新しい state machine / policy verdict を作らない。

---

## WS3 — ai-dev PR_CREATED Handoff Surface

`ai-dev-workflow` の Stable contract:

```text
Request -> Plan -> C-1 -> C-2 -> Human C-3
        -> Execute -> Verify -> PR_CREATED
```

を変更しない。

変更対象は `PR_CREATED` 時の presentation / handoff のみ。

候補:

```text
Changed
Verified
Open
Risks / Unknowns
Human Attention Required
```

---

## WS4 — Plan Decision Surface

**Blocked by #1337 result fixed.**

候補原則:

> Think deeply. Materialize selectively. Optimize the artifact for human decision-making.

Plan output候補:

```text
Goal
Evidence
Decision
Changes
Risks / Unknowns
Verification
Human Decision Required
```

#1337 baseline/candidate を固定後、第2candidateとしてpaired評価する。

---

## WS5 — Evolution Promotion Surface

#869 Evolution Loop と接続する。

```text
RunEvidence
 -> Retrospective
 -> Improvement Candidates
 -> Evaluation
 -> Promotion Surface
 -> Human Promotion Decision
```

Humanへ全候補をraw列挙しない。

表示カテゴリ例:

- PROMOTION_READY
- HUMAN_DECISION_REQUIRED
- OBSERVE
- REJECTED / INCONCLUSIVE

ただし Promotion authority / candidate evaluation semantics は変更しない。

---

## WS6 — Cross-layer Evaluation

Human Attentionを単独KPIにしない。

共通候補:

- Human Attention Time
- Time to Human Required
- Human Intervention Rate
- Decision Extraction Success
- Visibility Regression
- human reverse rate

Guard metrics:

- correctness
- critical visibility / recall
- false suppression
- provenance loss
- incomplete coverage visibility
- safety regression

---

# Dependency order

```text
WS0 Canon
   ↓
WS1 River Review proof
   ↓
WS2 ai-loop V2
   ├────────────→ WS5 Evolution
   ↓
WS3 ai-dev handoff

#1337 result fixed
   +
WS1 evidence
   ↓
WS4 Plan Decision Surface

WS1〜WS5
   ↓
WS6 cross-layer evaluation
   ↓
Harness Principle adoption / refinement
```

WS0 docsは River Review dogfood と並行可能。
WS4だけは #1337 完了を hard dependency とする。

---

# Rollout gates

各WSは一度にproduction contractへ入れない。

```text
Design
 -> Display-only / projection-only
 -> Dogfood
 -> Measure
 -> Independent review
 -> Adopt / revise / reject
```

新しい semantic Judge / Gate / state / artifact は、既存責務で表現できないことを証明した場合だけ検討する。

---

# Non-goals

- HumanをLoopから排除すること
- Human authorityの削減
- Legacy ai-loopへの新機能追加
- ai-dev public contractの変更
- hidden CoT保存
- raw transcriptのHuman表示
- fixed character limit
- top-Nだけを唯一の表示にすること
- attention metric単独最適化
- 新しいHuman Attention DB / memory subsystem
- 各repoへの同一実装コピー
- auto merge / Production Harness auto promotion

---

# Definition of Done

- [ ] WS0 Canon / Shared Principle
- [ ] River Reviewでbaseline/candidate dogfood完了
- [ ] Human Attention低下とvisibility/safety維持を同時確認
- [ ] ai-loop V2 Loop Decision Surfaceを実証
- [ ] ai-dev PR_CREATED handoffをstable contractを変えず改善
- [ ] #1337結果固定後にPlan Decision Surfaceをpaired評価
- [ ] Evolution Promotion Surfaceを#869へ接続
- [ ] cross-layer metricsを比較可能にする
- [ ] 各SurfaceがprojectionでありJudgeではないことを維持
- [ ] Human-owned authorityを維持
- [ ] 効果のないSurfaceを削除・簡素化できる

---

### PlanGate
- #870 ai-loop V2 parent Epic
- #869 Run Retrospective / Harness Evolution
- #1285 V2 RunEvidence metrics
- #1335 Plan Design Principles
- #1337 Plan Design Principles paired evaluation

### River Review
- s977043/river-review#2368
- s977043/river-review#2370
- s977043/river-review#1990
- s977043/river-review#2322

---

# Tracking

| WS | Issue | Status / dependency |
| --- | --- | --- |
| WS0 Canon | #1344 | **NEXT** |
| WS1 River Review proof | s977043/river-review#2368 | in progress (#2369 / #2370 / #2371) |
| WS2 ai-loop V2 | #1345 | blocked by WS0 + WS1 evidence |
| WS3 ai-dev handoff | #1346 | after WS0; reuse WS2 conventions where useful |
| WS4 Plan Decision Surface | #1347 | **hard blocked by #1337** + WS1 evidence |
| WS5 Evolution Promotion Surface | #1348 | after WS0 + WS2; connects #869 |
| WS6 Cross-layer evaluation | #1349 | evolves alongside WS1–WS5 |

## Immediate execution order

1. #1344 WS0 canon gap analysis / docs
2. Continue River Review #2368 dogfood
3. Freeze common metric glossary in #1349 without changing runtime schemas
4. Start #1345 after River Review evidence is available
5. #1346 after projection conventions stabilize
6. #1348 after V2 projection + #869 contracts align
7. #1347 only after #1337 result is fixed

**Comment by s977043** [comment]
WS0 #1344 を着手し、PR #1350 を作成しました。

現在:
- Parent Epic #1343: created
- WS0 #1344: PR #1350 open
- WS1 River Review #2368: in progress
- WS2 #1345: waiting for WS0 + River Review evidence
- WS3 #1346: queued
- WS4 #1347: hard blocked by #1337
- WS5 #1348: queued after WS2
- WS6 #1349: metric glossary can progress in parallel

#1350はdocs-onlyで、V2 North Starへ横断原則を追加する最小差分です。

**Comment by s977043** [comment]
Progress update:
- WS0 #1344: PR #1350, canon principle + review questions
- WS1 River Review #2368: ADR/renderer stack lint fixes applied; CI re-running
- WS6 #1349: PR #1356 created with cross-layer metric glossary + anti-gaming/privacy guards

Important review corrections:
1. Human Attention cannot be optimized alone.
2. Existing legacy metrics keep their existing meaning.
3. Visibility uses an independent material reference set, not baseline-as-oracle.
4. Human telemetry is not an individual performance KPI.

**Comment by s977043** [comment]
WS1 River Reviewの実証計画を具体化しました: s977043/river-review#2378

実装 #2371 merge後に、
1. fixed fixture baseline/candidate
2. material reference setによるvisibility検証
3. Decision Extraction / bounded attention measurement
4. 合格後のみ10〜20 PR dogfood

の順でEvidenceを作り、WS2 #1345へ渡します。

**Comment by s977043** [comment]
WS1 River Review evidence pipelineを具体化しました。

- river-review#2379: evaluation contract / fixtures / rubric
- river-review#2382: actual baseline/candidate commit paired runner

重要なreview hardening:
- baseline outputをoracleにしない
- signals→runtime resultのfixture adapterもfreeze対象
- baselineがcandidateのancestorであることを検証
- unrelated diffが混ざれば INCONCLUSIVE_SCOPE
- same underlying review state + same adapterでpresentationだけ比較

WS2 #1345はこのPhase A/B Evidenceが揃うまでblockedを維持します。

**Comment by s977043** [comment]
Current gate status:

- WS0 #1344 / PR #1350: all automated checks green; Human C-4 / merge pending
- WS1 River Review:
  - #2369 ADR: all checks green; Human C-4 pending
  - #2371 renderer: final CI validation in progress
  - #2379 eval contract: lint normalization applied; revalidation in progress
  - #2382 Phase A runner: hard blocked until #2369/#2371/#2379 accepted
- WS6 #1349 / PR #1356: all automated checks green; Human C-4 / merge pending

WS2 #1345 remains blocked until River Review Phase A/B Evidence exists.

**Comment by s977043** [comment]
WS1 evaluation planning progressed:

- River Review #2378: Phase A baseline/dogfood evaluation
- PR #2384: frozen 10-case fixture + extraction/visibility/timing protocol
- River Review #2385: execution harness/materialization slice

#2385はsame canonical input → baseline/candidate actual rendererの比較に限定し、Human judgmentやshared eval schemaは所有しません。WS2 #1345はこのEvidence完成まで開始しません。

**Comment by s977043** [comment]
Progress update:

- WS0 #1344 / PR #1350: all checks green, final review complete, Human C-4 pending
- WS6 #1349 / PR #1356: all checks green, final review complete, Human C-4 pending
- WS1 River Review:
  - #2369 ADR: all checks green
  - #2371 runtime: final tree squashed to one commit on #2369 base; exactly 4 files (runtime/test/dist), Prettier canonical + dist rebuilt; final CI re-running
  - #2378 / PR #2388: Phase A eval plan + machine-readable fixture oracle prepared; CI in progress

Important review corrections in this iteration:
- human-review-required != human decision required
- one timing sample cannot prove attention improvement
- counterbalanced sessions required when timing is used for adoption
- baseline is not the visibility oracle
- runtime PR history/diff boundary was cleaned before final validation.

**Comment by s977043** [comment]
Plan progress / review update:

- WS0 #1344: PR #1350 merged; issue closed.
- Parent execution order updated.
- River Review #2369 was rebased onto latest main as a 1-file / 1-commit ADR diff.
- #2371 was rebuilt on top of #2369 as a 1-commit / 4-file runtime-test-dist diff.
- An intermediate reparent attempt exposed a tree-reversion risk; it was detected by diff review and corrected using base-tree + explicit blob construction.
- #2388 now has a machine-readable 10-case oracle plus CI contract test.

Current gate remains: do not start #1345 until River Review Phase A evidence exists.

**Comment by s977043** [comment]
WS1 status advanced:

- River Review ADR #2369 merged
- Decision Surface #2371 merged
- Phase 1 issue #2370 closed
- Phase A evaluation contract #2379 merged
- #2398 now consolidates parallel #2378 evaluation assets into one SSoT before measurement
- #2382 paired runner is ready to start after #2398

WS2 #1345 remains correctly blocked until River Review effectiveness evidence exists. WS4 #1347 remains blocked by #1337, which is still INCONCLUSIVE_NOT_RUN.

**Comment by s977043** [comment]
## WS3 status update — #1346 Phase 0 design frozen

#1346 の Phase 0 設計を進め、Zenn記事由来の workflow Skillization / review decision 記録を現行 PlanGate に照合した。

### Fixed decisions

- 新しい `todays-work` 相当 monolithic Skill は作らない
- `workflow-conductor` + 既存 `ai-dev-*` Skill を維持
- 新しい `review-decisions.md` SSoT は作らない
- `review-external.md` / `decision-log.jsonl` / verification evidence から PR_CREATED Decision Surface を deterministic projection する
- 初期 rollout は presentation-only
- `PR_CREATED` / C-3 / C-4 / handoff 6要素 / schema / approval semantics は変更しない

### Current gate

River Review:
- #2369 accepted
- #2371 accepted
- #2378 evaluation in progress
- #2398 pending
- #2382 pending

したがって #1346 は **Phase 0 design = complete / runtime-template implementation = waiting for #2378 evidence** とする。

詳細設計・レビューは #1346 の最新コメント参照。

**Comment by s977043** [comment]
## WS3 evidence update

River Review Human Attention Phase A の deterministic layer が完了。

- frozen 10-case paired execution: PASS
- material visibility regression: 0
- candidate material miss: 0
- raw evidence preserved

ただし Human Decision Extraction / Human Attention は未評価で、candidateはinitial visible linesが最大 +5 行増えるケースあり。

したがって WS3 #1346 は:
- Phase 0 design: complete
- safety evidence: available
- Human Attention evidence: pending
- PlanGate runtime rollout: **still gated**

「安全に情報を前面化できる」までは確認できたが、「人間のattention costを下げる」はまだ確認していない。
